service/matching/taskReader.go

```diff
@@ -103,7 +103,7 @@ func (tr *taskReader) Signal() {
 }
 
 func (tr *taskReader) dispatchBufferedTasks(ctx context.Context) error {
-	ctx = headers.SetCallerInfo(ctx, headers.NewCallerInfo(headers.CallerTypeBackground))
+	ctx = tr.initContext(ctx)
```

Comment:
same change need to be applied to db_task_reader and db_task_writer. cc @maoyi16

Reply:
Done.